lib/mux/webhooks.ex

```diff
@@ -5,6 +5,38 @@ defmodule Mux.Webhooks do
   @default_tolerance 300
   @expected_scheme "v1"
 
+  @doc """
+  Generates a webhook signature. Pass in the raw request body and the
+  webhook "secret". You may also pass in the desired signature scheme.
+  Currently, the only valid signature scheme is `"v1"`, which is also the
+  default one if not set explicitly. Any invalid signature scheme will fall
+  back to the default.
+
+  Note that this function is intended for testing against malformed or
+  malicious webhook requests, so you should use a fake webhook "secret".
+
+  Returns the relevant HTTP header value as a string.
+
+  ## Examples
+
+      iex> signature = Mux.Webhooks.generate_signature(payload, fake_secret)
+      "t=1565125718,v1=854ece4c22acef7c66b57d4e504153bc512595e8e9c772ece2a68150548c19a7"
+
+  You can then verify the header, e.g.:
+
+      iex> Mux.Webhooks.verify_header(payload, signature, fake_secret)
+      :ok
+  """
+  def generate_signature(payload, secret, _scheme \\ @expected_scheme) do
```

Comment:
I think I'm okay with exposing this, but let me give it a little more thought.

The use case is:

  * You're writing tests for your elixir app
  * You want to generate valid signatures and invalid signatures in your test environment

Exposing this function makes that easier to do.

Reply:
Indeed, that's the use case I had in mind.

Reply:
Revisiting this. I can see how it's useful for testing. One concern that came up internally is that having this in `Mux.Webhooks` module and in the documentation there might confuse some users who think they they should be using it to generate headers.

Since this would never actually be used in application code, I would like to make that more clear by putting it in a new module. Something like `Mux.Webhooks.TestUtils.generate_signature`

What do you think about making that change?

Reply:
That's a good point, I agree that having it in `Mux.Webhooks.TestUtils` would clarify things. I can get it done in a week, sounds good?